internal/ask/form/aggregation.go

```diff
@@ -0,0 +1,429 @@
+package form
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+
+	"github.com/ardanlabs/kit/log"
+	"github.com/coralproject/shelf/internal/ask/form/submission"
+	"github.com/coralproject/shelf/internal/platform/db"
+	"gopkg.in/mgo.v2/bson"
+)
+
+//==============================================================================
+
+// MCAnswerAggregation holds the count for selections of a single multiple
+// choice answer.
+type MCAnswerAggregation struct {
+	Title string `json:"answer" bson:"answer"`
+	Count int    `json:"count" bson:"count"`
+}
+
+// MCAggregation holds a multiple choice question and a map aggregated counts for
+// each answer. The Answers map is keyed off an md5 of the answer as not better keys exist
+type MCAggregation struct {
+	Question  string                         `json:"question" bson:"question"`
+	MCAnswers map[string]MCAnswerAggregation `json:"answers" bson:"answers"`
+}
+
+// TextAggregation holds the aggregated text based answers for a single question
+// marked with the Incude in Aggregations tag, orderd by [question_id][answer].
+type TextAggregation map[string]string
+
+// Aggregation holds the various aggregations and stats collected.
+type Aggregation struct {
+	Group Group                    `json:"group" bson:"group"`
+	Count int                      `json:"count" bson:"count"`
+	MC    map[string]MCAggregation `json:"MultipleChoice" bson:"MultipleChoice"` // Capitalization matches widget type MultipleChoice
+}
+
+// Group defines a key for a multiple choice question / answer combo to be used
+// to define slices of submissions to be aggregated.
+type Group struct {
+	ID       string `json:"group_id" bson:"group_id"`
+	Question string `json:"question" bson:"question"`
+	Answer   string `json:"answer" bson:"answer"`
+}
+
```

Comment:
Why not put these types under under a model and have validation for them?  Is there a reason we aren't validating these?